src/aws-redshiftserverless/workgroup.ts

```diff
@@ -0,0 +1,357 @@
+    };
+
+    this.validateCapacity();
+    this.validateWorkgroupName();
+    this.validatePort();
+    this.validateSubnet();
+
+    const workgroup = this.createWorkgroup(this, 'Resource', {
+      baseCapacity: this.props.baseCapacity,
+      configParameters: this.props.configParameters
+        ? Object.entries(this.props.configParameters).map(([key, value]) => ({
+            parameterKey: key,
+            parameterValue: value,
+          }))
+        : undefined,
+      enhancedVpcRouting: this.props.enhancedVpcRouting,
+      namespaceName: this.props.namespace?.namespaceName,
+      publiclyAccessible: this.props.publiclyAccessible,
+      port: this.props.port,
+      securityGroupIds: Lazy.list({ produce: () => this.securityGroups.map(sg => sg.securityGroupId) }),
+      subnetIds: Lazy.list({ produce: () => this.props.vpc.selectSubnets(this.vpcSubnets).subnetIds }),
+      workgroupName: this.physicalName,
+    });
+
+    this.workgroupArn = workgroup.attrWorkgroupWorkgroupArn;
+    this.workgroupName = workgroup.attrWorkgroupWorkgroupName;
+    this.workgroupId = workgroup.attrWorkgroupWorkgroupId;
+    this.endpointAddress = workgroup.attrWorkgroupEndpointAddress;
+    this.port = workgroup.attrWorkgroupEndpointPort;
+  }
+
+  protected createWorkgroup(
+    scope: Construct,
+    id: string,
+    props: aws_redshiftserverless.CfnWorkgroupProps,
+  ): aws_redshiftserverless.CfnWorkgroup {
+    return new aws_redshiftserverless.CfnWorkgroup(scope, id, props);
+  }
+
+  protected createSecurityGroup(
+    scope: Construct,
+    id: string,
+    props: aws_ec2.SecurityGroupProps,
+  ): aws_ec2.SecurityGroup {
+    return new aws_ec2.SecurityGroup(scope, id, props);
+  }
+
+  /**
+   * Validates capacity settings.
+   */
+  private validateCapacity(): void {
+    const baseCapacity = this.props.baseCapacity;
+
+    if (!Token.isUnresolved(baseCapacity) && baseCapacity !== undefined) {
+      if (baseCapacity < 8 || baseCapacity > 1024) {
+        throw new Error(`\`baseCapacity\` must be between 8 and 1024, got: ${baseCapacity}.`);
+      }
+
+      if (8 <= baseCapacity && baseCapacity <= 512 && baseCapacity % 8 !== 0) {
+        throw new Error(`\`baseCapacity\` must be units of 8 between 8 and 512, got: ${baseCapacity}.`);
+      }
+
+      if (512 <= baseCapacity && baseCapacity <= 1024 && baseCapacity % 32 !== 0) {
+        throw new Error(`\`baseCapacity\` must be units of 32 between 512 and 1024, got: ${baseCapacity}.`);
+      }
+    }
+  }
+
+  /**
+   * Validates a workgroup name.
+   */
+  private validateWorkgroupName(): void {
+    const workgroupName = this.props.workgroupName;
+    if (Token.isUnresolved(workgroupName) || workgroupName === undefined) {
+      return;
+    }
+
+    if (!/^[a-z0-9-]{3,64}$/.test(workgroupName)) {
+      throw new Error(
+        `\`workgroupName\` must be between 3 and 64 characters long, contain only lowercase letters, numbers, and hyphens, got: ${workgroupName}.`,
+      );
```

Comment:
It would be good if JSDoc had an explanation.

Reply:
Added to props JSDoc.